### PR TITLE
【KernelGen】Add top_k_per_row_prefill operator for vLLM speculative decoding

### PR DESCRIPTION
**Pull request by Schopenhauer-loves-Hegel**
Implement per-row top-k selection for prefill phase using Triton kernels.

### PR Category
Operator

### Type of Change
New Feature

### Description
Add `top_k_per_row_prefill` operator for vLLM's speculative decoding prefill phase. This operator selects top-k indices from variable-length rows of logits tensor.

Key features:
- Supports variable-length rows with row_starts and row_ends parameters
- Uses radix-based histogram algorithm for efficient O(n) selection
- Float32 to uint32 order-preserving conversion for histogram binning
- 4-round refinement using 8 bits per round (32 bits total)
- Handles boundary cases properly (valid_count < K)

### Issue
N/A

### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [x] Change is fully covered by a UT.

### Performance
benchmark/test_vllm_perf.py::test_top_k_per_row_prefill_benchmark
Operator: top_k_per_row_prefill  Performance Test (dtype=small_rows, mode=kernel,level=comprehensive)
Status       Torch Latency (ms)    Gems Latency (ms)         Gems Speedup          Size Detail
-----------------------------------------------------------------------------------------------
SUCCESS               0.021664            0.126848               0.171          [torch.Size([32, 20000]), torch.Size([32]), torch.Size([32]), torch.Size([32, 2048]), 32, 20000, 1, 2048]
SUCCESS               0.023072            0.130240               0.177          [torch.Size([128, 20000]), torch.Size([128]), torch.Size([128]), torch.Size([128, 2048]), 128, 20000, 1, 2048]

Operator: top_k_per_row_prefill  Performance Test (dtype=medium_rows, mode=kernel,level=comprehensive)
Status       Torch Latency (ms)    Gems Latency (ms)         Gems Speedup          Size Detail
-----------------------------------------------------------------------------------------------
SUCCESS               0.085472            0.441920               0.193          [torch.Size([1024, 20000]), torch.Size([1024]), torch.Size([1024]), torch.Size([1024, 2048]), 1024, 20000, 1, 2048]
SUCCESS               0.146240            0.834176               0.175          [torch.Size([2048, 20000]), torch.Size([2048]), torch.Size([2048]), torch.Size([2048, 2048]), 2048, 20000, 1, 2048]

Operator: top_k_per_row_prefill  Performance Test (dtype=large_rows, mode=kernel,level=comprehensive)
Status       Torch Latency (ms)    Gems Latency (ms)         Gems Speedup          Size Detail
-----------------------------------------------------------------------------------------------
SUCCESS               0.256960            1.562352               0.164          [torch.Size([4096, 20000]), torch.Size([4096]), torch.Size([4096]), torch.Size([4096, 2048]), 4096, 20000, 1, 2048]
SUCCESS               0.478496            3.028256               0.158          [torch.Size([8192, 20000]), torch.Size([8192]), torch.Size([8192]), torch.Size([8192, 2048]), 8192, 20000, 1, 2048]

PASSED

### CODE REVIEW

**Review by tengqm — APPROVED** [review]

**Comment by bin913** [comment]
@Schopenhauer-loves-Hegel Please resolve conflicts

**Comment by bin913** [comment]
@Schopenhauer-loves-Hegel Do you still keep track of this PR?
